preprocessor.py

```diff
@@ -234,10 +235,16 @@ def __init__(self, source_page_id, target_page_id, order):
 # export user catalog
 if config['User']['export']:
 
-    us=natsorted(list(set(list(map(lambda x: str(x['_id'])+'\n',recdb["user"].find({}))))),alg=ns.ns.SIGNED)
+    # produce users csv with each user id along with the user's accessed services
+    # query users from database for fields _id and accessed_services then create a list of rows
+    # each rows contains two elements, first: user_id in string format and second: a space separated sorted list of accessed services 
+    users = recdb['user'].find({},{'accessed_services':1})
```

Comment:
An issue is that it breaks the optionality of the --users flag, because the accesses services can only be retrieved through the Source (Mongo DB) and not from the user_actions.csv.

Reply:
In order to have hit rate I need to have specific user information